src/Agent.Sdk/ProcessInvoker.cs

```diff
@@ -805,6 +810,54 @@ private void NixKillProcessTree()
             }
         }
 
+        private void WriteProcessOomScoreAdj(Process proc)
+        {
+            try
+            {
+                if (proc.StartInfo.FileName.Contains("Agent.Worker", StringComparison.OrdinalIgnoreCase) ||
+                    proc.StartInfo.FileName.Contains("Agent.PluginHost", StringComparison.OrdinalIgnoreCase))
+                {
+                    return;
+                }
+                string procFilePath = $"/proc/{proc.Id}/oom_score_adj";
+                // Hosted Ubuntu mounts procfs at /proc, NOP on other platforms
+                if (File.Exists(procFilePath))
+                {
+                    if (proc.StartInfo.Environment.ContainsKey("VSTS_JOB_OOMSCOREADJ"))
+                    {
+                        string userOomScoreAdj = proc.StartInfo.Environment["VSTS_JOB_OOMSCOREADJ"];
+                        File.WriteAllText(procFilePath, userOomScoreAdj);
+                    }
+                    else
+                    {
+                        // Existing score is inherited from parent proc
+                        int oomScoreAdjExisting = int.Parse(File.ReadAllText(procFilePath));
```

Comment:
why can't we just set to 1000?

Reply:
As discussed, setting to 500 is sufficiently high and leaves room for Linux system admins to set other processes as less important than their pipeline jobs